modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -153,76 +334,71 @@ public static void extractQueryTerms(Query query, ParseContext.Document document
                         continue;
                     }
 
-                    Set<Term> temp;
+                    Result temp;
                     try {
                         temp = extractQueryTerms(clause.getQuery());
+                        if (temp.noTerms) {
+                            continue;
```

Comment:
temp.noTerms && temp.verified? otherwise we would have issues with MUST clauses containing match_no_docs queries?